File: frontend/src/app/services/carro.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { categoria } from '../interfaces/categoria';
import { cor } from '../interfaces/cor';
import { fabricante } from '../interfaces/fabricante';
import { combustivel } from '../interfaces/combustivel';
import { carro } from '../interfaces/carro';

@Injectable({
    providedIn: 'root'
})

export class carroService {

    private apiUrl = 'http://localhost:8080/carro';

    constructor(private http: HttpClient) { }

    cadastrarCarro(carro: carro) {
        return this.http.post(`${this.apiUrl + "/cadastrarCarro"}`, carro);
    }

    editarCarro(id: number, carro: carro) {
        return this.http.put(`${this.apiUrl}/editarCarro/${id}`, carro);
    }

    deletarCarro(id: number): Observable<carro> {
        return this.http.delete<carro>(`${this.apiUrl}/deletarCarro/${id}`);
    }

    buscarCarroPorId(id: number) {
        return this.http.get<carro>(`${this.apiUrl}/public/buscarCarroPorId/${id}`);
    }

    listarCarro(): Observable<carro[]> {
        return this.http.get<carro[]>(`${this.apiUrl}/listarCarro`)
    }

    listarCategoria(): Observable<categoria[]> {
        return this.http.get<categoria[]>(`${this.apiUrl}/public/listarCategoria`);
    }

    listarCor(): Observable<cor[]> {
        return this.http.get<cor[]>(`${this.apiUrl}/public/listarCor`);
    }

    listarFabricante(): Observable<fabricante[]> {
        return this.http.get<fabricante[]>(`${this.apiUrl}/public/listarFabricante`);
    }

    listarCombustivel(): Observable<combustivel[]> {
        return this.http.get<combustivel[]>(`${this.apiUrl}/public/listarCombustivel`);
    }
}

File: backend-v2/src/main/java/projeto/backend/Services/CarroService.java
package projeto.backend.Services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import projeto.backend.Entities.*;
import projeto.backend.Repositories.*;

import java.util.List;
import java.util.Optional;

@Service
public class CarroService {

    @Autowired
    private CategoriaRepository categoriaRepository;
    @Autowired
    private CorRepository corRepository;
    @Autowired
    private FabricanteRepository fabricanteRepository;
    @Autowired
    private CombustivelRepository combustivelRepository;
    @Autowired
    private CarroRepository carroRepository;

    public List<Categoria> listarCategoria() {
        return categoriaRepository.findAll();
    }

    public List<Cor> listarCor() {
        return corRepository.findAll();
    }

    public List<Fabricante> listarFabricante() {
        return fabricanteRepository.findAll();
    }

    public List<Combustivel> listarCombustivel() {
        return combustivelRepository.findAll();
    }

    public Optional<Carro> buscarCarroPorPlaca(String placa) {
        return carroRepository.findByPlaca(placa);
    }

    public Optional<Carro> buscarCarroPorId(Long id){
        return carroRepository.findById(id);
    }

    public List<Carro> listarCarro(){
        return  carroRepository.findAll();
    }

    public Carro cadastrarCarro(Carro carro){
        return carroRepository.save(carro);
    }

    public void deletarCarro(Long id){
        carroRepository.deleteById(id);
    }
}
